File: links.ts
import { ReactNode } from "react";
import { Tux, Windows, FreeBSD, Apple } from "./components/logos";

export const discord = "https://discord.gg/rabs2GZd8b";
export const github = "https://github.com/chatterino/chatterino2";
export const tip = "https://streamelements.com/fourtf/tip";
export const wiki = "https://wiki.chatterino.com";
export const nightly =
  "https://github.com/Chatterino/chatterino2/releases/tag/nightly-build";
export const allDownloads = "download";

export const currentVersion = "2.5.4";
export const baseDownloadLink = `https://chatterino.fra1.digitaloceanspaces.com/bin`;
const dl = `https://chatterino.fra1.digitaloceanspaces.com/bin/${currentVersion}`;

export const allVersions = [
  "2.1.0",
  "2.1.4",
  "2.1.5-4",
  "2.1.6",
  "2.1.7",
  "2.2.0",
  "2.2.1",
  "2.2.2-fix",
  "2.2.3-beta2",
  "2.3.0",
  "2.3.1",
  "2.3.2",
  "2.3.3",
  "2.3.4",
  "2.3.5",
  "2.4.0",
  "2.4.1",
  "2.4.2",
  "2.4.3",
  "2.4.4",
  "2.4.5",
  "2.4.6",
  "2.5.0",
  "2.5.1",
  "2.5.2",
];
export const allV1Versions = [
  "0.2.6.4",
  "1.0.9",
  "1.1.0.0",
  "1.2.11",
  "1.2.12",
  "1.2.13",
  "1.2.3.0",
  "1.2.4",
  "1.2.5",
  "1.2.6",
  "1.2.8",
  "1.2.9",
  "1.3.2",
  "1.3",
];

export type DownloadType = [() => React.JSX.Element, string, string];
export const windows: DownloadType = [
  Windows,
  `${currentVersion} for Windows 64-Bit`,
  `${dl}/Chatterino.Installer.exe`,
];
export const linux: DownloadType = [Tux, "Linux", "linux"];
export const freeBsd: DownloadType = [FreeBSD, "FreeBSD", "freebsd"];
export const macOs: DownloadType = [Apple, "macOS", `${dl}/Chatterino.dmg`];
export const windowsPortable = `${dl}/Chatterino.Portable.zip`;
export const linuxAppimageUrl = `${dl}/Chatterino-x86_64.AppImage`;

export const linuxBuildFromSource =
  "https://github.com/Chatterino/chatterino2/blob/master/BUILDING_ON_LINUX.md";

File: src/links.rs
use std::fs;
use std::path::Path;

use anyhow::{Context, Result};
use serde::{Deserialize, Serialize};

// #[derive(Deserialize, Serialize)]
// pub struct Links {
//     pub discord: String,
//     pub github: String,
//     pub tip: String,
//     pub wiki: String,
//     pub nightly: String,
//     pub all_downloads: String,
//     pub current_version: String,
//     pub base_download_link: String,
//     pub linux_build_from_source: String,
//     pub versions: Vec<String>,
// }

// /// Load `links.json`, which contains various links, base urls, and also c2 versions.
// pub fn load(path: &Path) -> Result<Links> {
//     let source =
//         fs::read_to_string(path).with_context(|| format!("failed to read {}", path.display()))?;
//     serde_json::from_str(&source).with_context(|| format!("failed to parse {}", path.display()))
// }

// #[derive(Deserialize, Serialize)]
// pub struct Download {
//     pub icon: String,
//     pub label: String,
//     pub href: String,
// }

// #[derive(Serialize)]
// pub struct Downloads {
//     pub windows: Download,
//     pub linux: Download,
//     pub macos: Download,
//     pub freebsd: Download,
//     pub windows_portable: String,
// }

// impl Downloads {
//     pub fn new(links: &Links) -> Downloads {
//         let version = &links.current_version;
//         let dl = format!("{}/{version}", links.base_download_link);
//         Downloads {
//             windows: Download {
//                 icon: "windows".to_owned(),
//                 label: format!("{version} for Windows 64-Bit"),
//                 href: format!("{dl}/Chatterino.Installer.exe"),
//             },
//             linux: Download {
//                 icon: "tux".to_owned(),
//                 label: "Linux".to_owned(),
//                 href: "linux".to_owned(),
//             },
//             macos: Download {
//                 icon: "apple".to_owned(),
//                 label: "macOS".to_owned(),
//                 href: format!("{dl}/Chatterino.dmg"),
//             },
//             freebsd: Download {
//                 icon: "freebsd".to_owned(),
//                 label: "FreeBSD".to_owned(),
//                 href: "freebsd".to_owned(),
//             },
//             windows_portable: format!("{dl}/Chatterino.Portable.zip"),
//         }
//     }
// }
